src/Workspaces/Core/Portable/Shared/Extensions/SemanticModelExtensions.cs

```diff
@@ -176,16 +176,34 @@ private static ISymbol MapSymbol(ISymbol symbol, ITypeSymbol type)
             SyntaxToken token,
             CancellationToken cancellationToken)
         {
-            var aliasSymbol = semanticModel.GetAliasInfo(token.Parent, cancellationToken);
+            IAliasSymbol aliasSymbol;
+            ITypeSymbol type;
+            ISymbol declaredSymbol;
+            ImmutableArray<ISymbol> allSymbols;
 
-            var bindableParent = syntaxFacts.GetBindableParent(token);
-            var type = semanticModel.GetTypeInfo(bindableParent, cancellationToken).Type;
-
-            var declaredSymbol = MapSymbol(semanticFacts.GetDeclaredSymbol(semanticModel, token, cancellationToken), type);
-            var allSymbols = semanticModel.GetSymbolInfo(bindableParent, cancellationToken)
-                                          .GetBestOrAllSymbols()
-                                          .WhereAsArray(s => !s.Equals(declaredSymbol))
-                                          .SelectAsArray(s => MapSymbol(s, type));
+            var overriddingIdentifier = syntaxFacts.GetDeclarationIdentifierIfOverride(token);
+            if (overriddingIdentifier.HasValue)
+            {
+                // on an "override" token, we'll find the overridden symbol
+                aliasSymbol = null;
+                var overriddingSymbol = semanticFacts.GetDeclaredSymbol(semanticModel, overriddingIdentifier.Value, cancellationToken);
+                var overriddenSymbol = overriddingSymbol.GetOverriddenMember();
+                type = overriddenSymbol as ITypeSymbol;
```

Comment:
isn't this always null?

Reply:
You're right. For some reason I was thinking of base types, but those aren't overrides.